Modules/Registration/Metricsv4/test/itkEuclideanDistancePointSetMetricTest3.cxx

```diff
@@ -0,0 +1,220 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkEuclideanDistancePointSetToPointSetMetricv4.h"
+#include "itkTranslationTransform.h"
+#include "itkTestingMacros.h"
+
+#include <fstream>
+#include "itkMath.h"
+
+/*
+ * Test with a displacement field transform
+ */
+
+template <unsigned int Dimension>
+int
+itkEuclideanDistancePointSetMetricTest3Run()
+{
+  using PointSetType = itk::PointSet<float, Dimension>;
+  using PointType = typename PointSetType::PointType;
+
+  using PointsContainerType = itk::VectorContainer<long unsigned int, PointType>;
+  using PointsLocatorType = itk::PointsLocator<PointsContainerType>;
+  auto pointsLocator = PointsLocatorType::New();
+
+  auto fixedPoints = PointSetType::New();
+  fixedPoints->Initialize();
+
+  auto movingPoints = PointSetType::New();
+  movingPoints->Initialize();
+
+  // Threshold used to test the distance threshold functionality
+  double distanceThreshold = 0.5;
```

Comment:
Are all the added `if/else` cases concerning the distance threshold value exercised in this test? Maybe the ones not covered here are covered in other tests through the default value. But for the sake of completeness, I would make sure all are tested here, maybe by making the value of the threshold (and the expected output values, if necessary) be a parameter of the method.

Reply:
:100: Thanks @PranjalSahu.